### ISSUE TITLE
farmhash-0.1.0.1 can't link its test suite

### ISSUE THREAD

**Issue by peti** [abhinav/farmhash#1]
http://hydra.cryp.to/build/610012/log/raw:

```
Linking dist/build/test-farmhash/test-farmhash ...
/tmp/nix-build-haskell-farmhash-0.1.0.1.drv-0/farmhash-0.1.0.1/dist/build/libHSfarmhash-0.1.0.1-ghc7.8.4.so: undefined reference to `__gxx_personality_v0'
```

**Comment by abhinav** [comment]
I can't reproduce this locally. The tests link and run successfully on both, my laptop and on the NixOS VM I set up specifically to try to reproduce this issue. I ran `nix-build` on it both, with and without `--pure`.

```
[vagrant@nixbox64:/farmhash]$ nix-build -E '(import <nixpkgs> {}).callPackage ./shell.nix {}' --pure
download-from-binary-cache.pl: still waiting for ‘http://cache.nixos.org/cx11m2kasd7aqnwl9ky3k9frzawkl80i.narinfo’ after 5 seconds...
these derivations will be built:
  /nix/store/l9qqbq73akvjnfiq3iarjjkcrxv1ycra-haskell-farmhash-ghc7.6.3-0.1.0.1.drv
building path(s) `/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1'
building /nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1
unpacking sources
unpacking source archive /nix/store/00l11lmfyrq63chb7w46pj9gzwmadqsx-farmhash
source root is farmhash
patching sources
configuring
[1 of 1] Compiling Main             ( Setup.hs, Setup.o )
Linking Setup ...
configure flags: --enable-split-objs --disable-library-profiling --disable-shared --enable-library-vanilla --disable-executable-dynamic --enable-tests
Configuring farmhash-0.1.0.1...
Dependency QuickCheck ==2.7.*: using QuickCheck-2.7.6
Dependency base >=4.6 && <4.8: using base-4.6.0.1
Dependency bytestring -any: using bytestring-0.10.0.2
Dependency farmhash -any: using farmhash-0.1.0.1
Dependency hspec ==2.1.*: using hspec-2.1.4
Using Cabal-1.16.0 compiled by ghc-7.6
Using compiler: ghc-7.6.3
Using install prefix:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1
Binaries installed in:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/bin
Libraries installed in:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/farmhash-0.1.0.1
Private binaries installed in:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/libexec
Data files installed in:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/share/farmhash-0.1.0.1
Documentation installed in:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/share/doc/farmhash-0.1.0.1
No alex found
Using ar found on system at:
/nix/store/r1ajivbpa0gfd0yi7gijdsv87gpz972x-binutils-2.23.1/bin/ar
No c2hs found
No cpphs found
No ffihugs found
Using gcc version 4.8.3 found on system at:
/nix/store/d44lppqijrvscv6y0ynhv5ybka32658r-gcc-wrapper-4.8.3/bin/gcc
Using ghc version 7.6.3 found on system at:
/nix/store/f3axa43yppcak7vvw628abjzs96xplsw-ghc-7.6.3-wrapper/bin/ghc
Using ghc-pkg version 7.6.3 found on system at:
/nix/store/f3axa43yppcak7vvw628abjzs96xplsw-ghc-7.6.3-wrapper/bin/ghc-pkg
No greencard found
Using haddock version 2.13.2 found on system at:
/nix/store/29smpa0qfg2qzg1ycvaf57xsibr66swy-ghc-7.6.3/bin/haddock
No happy found
No hmake found
Using hpc version 0.6 found on system at:
/nix/store/f3axa43yppcak7vvw628abjzs96xplsw-ghc-7.6.3-wrapper/bin/hpc
Using hsc2hs version 0.67 found on system at:
/nix/store/f3axa43yppcak7vvw628abjzs96xplsw-ghc-7.6.3-wrapper/bin/hsc2hs
No hscolour found
No hugs found
No jhc found
Using ld found on system at:
/nix/store/d44lppqijrvscv6y0ynhv5ybka32658r-gcc-wrapper-4.8.3/bin/ld
No lhc found
No lhc-pkg found
No nhc98 found
No pkg-config found
Using ranlib found on system at:
/nix/store/r1ajivbpa0gfd0yi7gijdsv87gpz972x-binutils-2.23.1/bin/ranlib
Using strip found on system at:
/nix/store/r1ajivbpa0gfd0yi7gijdsv87gpz972x-binutils-2.23.1/bin/strip
Using tar found on system at:
/nix/store/gbwlpj5wlaaxc3fsxwnhjdqva8w3dlmr-gnutar-1.27.1/bin/tar
No uhc found
building
Building farmhash-0.1.0.1...
Preprocessing library farmhash-0.1.0.1...
[1 of 2] Compiling FarmHash.Internal ( src/FarmHash/Internal.hs, dist/build/FarmHash/Internal.o )
[2 of 2] Compiling FarmHash         ( src/FarmHash.hs, dist/build/FarmHash.o )
In-place registering farmhash-0.1.0.1...
Preprocessing test suite 'test-farmhash' for farmhash-0.1.0.1...
[1 of 2] Compiling FarmHashSpec     ( test/FarmHashSpec.hs, dist/build/test-farmhash/test-farmhash-tmp/FarmHashSpec.o )
[2 of 2] Compiling Main             ( test/Spec.hs, dist/build/test-farmhash/test-farmhash-tmp/Main.o )
Linking dist/build/test-farmhash/test-farmhash ...
Running Haddock for farmhash-0.1.0.1...
Preprocessing library farmhash-0.1.0.1...
Warning: The documentation for the following packages are not installed. No
links will be generated to these packages: rts-1.0
Haddock coverage:
   0% (  0 / 10) in 'FarmHash.Internal'
 100% ( 13 / 13) in 'FarmHash'
Documentation created: dist/doc/html/farmhash/index.html,
dist/doc/html/farmhash/farmhash.txt
Preprocessing test suite 'test-farmhash' for farmhash-0.1.0.1...
running tests
Running 1 test suites...
Test suite test-farmhash: RUNNING...
Test suite test-farmhash: PASS
Test suite logged to: dist/test/farmhash-0.1.0.1-test-farmhash.log
1 of 1 test suites (1 of 1 test cases) passed.
installing
Installing library in
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/farmhash-0.1.0.1
Creating package registration file:
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/package.conf.d/farmhash-0.1.0.1.conf
Reading package info from "/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/package.conf.d/farmhash-0.1.0.1.conf" ... done.
farmhash-0.1.0.1: dependency "base-4.6.0.1-02ac91ff7de681afee6a8fa62f0c87f3" doesn't exist (ignoring)
farmhash-0.1.0.1: dependency "bytestring-0.10.0.2-af9d348ce49438db8d6083fb479dbb53" doesn't exist (ignoring)
Writing new package config file... done.
post-installation fixup
gzipping man pages
stripping (with flags -S) in  /nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib  /nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/bin
strip:/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/farmhash-0.1.0.1/FarmHash/Internal.hi: File format not recognized
strip:/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/farmhash-0.1.0.1/FarmHash.hi: File format not recognized
strip:/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/package.conf.d/farmhash-0.1.0.1.conf: File format not recognized
strip:/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1/lib/ghc-7.6.3/package.conf.d/farmhash-0.1.0.1.installedconf: File format not recognized
patching ELF executables and libraries
patching script interpreter paths
/nix/store/cx11m2kasd7aqnwl9ky3k9frzawkl80i-haskell-farmhash-ghc7.6.3-0.1.0.1
```

(I wasn't sure how to get my NixOS 14.04 to use the most recent nixpkgs.)

Is this possibly an issue on the build server's side?

**Comment by peti** [comment]
> I wasn't sure how to get my NixOS 14.04 to use the most recent nixpkgs.

Check out the `master` branch of Nixpkgs. Then pass `-I nixpkgs=/location` to the `nix-build` command line.

> Is this possibly an issue on the build server's side?

I can think of an issue the server might have that triggers this error? Even so, the server is running the latest NixOS, so if the sever has that issue, then every other NixOS machine following the master branch will have the same issue, so it would be nice to address it in either case.

**Comment by abhinav** [comment]
Managed to reproduce. Fixed and released in 0.1.0.2.
